ide/html.editor/src/org/netbeans/modules/html/editor/hints/other/AddMissingAltAttributeRule.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.html.editor.hints.other;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.Document;
+import org.netbeans.api.html.lexer.HTMLTokenId;
+import org.netbeans.api.lexer.Token;
+import org.netbeans.api.lexer.TokenHierarchy;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.api.project.FileOwnerQuery;
+import org.netbeans.api.project.Project;
+import org.netbeans.modules.csl.api.Hint;
+import org.netbeans.modules.csl.api.HintSeverity;
+import org.netbeans.modules.csl.api.OffsetRange;
+import org.netbeans.modules.csl.api.RuleContext;
+import org.netbeans.modules.html.editor.api.Utils;
+import org.netbeans.modules.html.editor.api.gsf.HtmlParserResult;
+import org.netbeans.modules.html.editor.hints.HtmlRule;
+import org.netbeans.modules.html.editor.hints.HtmlRuleContext;
+import org.netbeans.modules.html.editor.refactoring.InlinedStyleInfo;
+import org.netbeans.modules.html.editor.utils.HtmlTagContextUtils;
+import org.netbeans.modules.web.common.api.WebUtils;
+import org.openide.filesystems.FileObject;
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author Chris
+ */
+@NbBundle.Messages("AddMissingAltAttributeRuleName=Insert required 'alt' attribute")
```

Comment:
This message key should be `AddMissingAltAttributeRule` - you can then drop the implementation of `getDisplayName` as the superclass already implements it:

https://github.com/apache/netbeans/blob/a8edf3bf92e38ac22a83e24ec910f29094f0fee3/ide/html.editor/src/org/netbeans/modules/html/editor/hints/HtmlRule.java#L86-L89
with
https://github.com/apache/netbeans/blob/a8edf3bf92e38ac22a83e24ec910f29094f0fee3/ide/html.editor/src/org/netbeans/modules/html/editor/hints/HtmlRule.java#L60-L63

You also need to define a message named `AddMissingAltAttributeRule_Desc` (you get an exception when you click on the hint in the option panel):

https://github.com/apache/netbeans/blob/a8edf3bf92e38ac22a83e24ec910f29094f0fee3/ide/html.editor/src/org/netbeans/modules/html/editor/hints/HtmlRule.java#L65-L68

I suggest to use : `Provide Text Alternatives` for the display name (to get it more in line with the rest of the entries) and `Find img/applet/area elements where no alt attribute is provided.` (for the description)

![image](https://user-images.githubusercontent.com/2179736/208514700-8d7ab9fe-ea80-4fe2-a993-74aea0cbb743.png)